include/core/Godot.hpp

```diff
@@ -246,6 +246,12 @@ void register_method(const char *name, M method_ptr, godot_method_rpc_mode rpc_t
 	godot::nativescript_api->godot_nativescript_register_method(godot::_RegisterState::nativescript_handle, ___get_method_class_name(method_ptr), name, attr, method);
 }
 
+// User can specify a derived class D to register the method for, instead of it being inferred.
+template <class D, class B, class R, class... As>
+void register_class_method(const char *name, R (B::*method_ptr)(As...), godot_method_rpc_mode rpc_type = GODOT_METHOD_RPC_MODE_DISABLED) {
```

Comment:
I'd probably prefer a name like `register_method_explicit`  instead since in some programming circles a "class method" can mean a static method.

I'm not sure about the exact naming, maybe there is something better than `register_method_explicit`, open to suggestions :)

Reply:
`register_method_explicit` sounds fine to me